src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.SetTimes.OSX.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.IO
+{
+    internal partial struct FileStatus
+    {
+        internal void SetCreationTime(string path, DateTimeOffset time)
+        {
+            // Try to set the attribute on the file system entry using setattrlist,
+            // if we get ENOTSUP then it means that "The volume does not support
+            // setattrlist()", so we fall back to the method used on other unix
+            // platforms, otherwise we throw an error if we get one, or invalidate
+            // the cache if successful because otherwise it has invalid information.
+            // Note: the unix fallback implementation doesn't have a test as we are
+            // yet to determine which volume types it can fail on, so modify with
+            // great care.
+            Interop.Error error = SetCreationTimeCore(path, time);
+
+            if (error == Interop.Error.SUCCESS)
+            {
+                InvalidateCaches();
+            }
+            else if (error == Interop.Error.ENOTSUP)
+            {
+                SetAccessOrWriteTimeCore(path, time, isAccessTime: false, checkCreationTime: false);
+            }
+            else
+            {
+                Interop.CheckIo(error, path, InitiallyDirectory);
+            }
+        }
+
+        private unsafe Interop.Error SetCreationTimeCore(string path, DateTimeOffset time)
+        {
+            Interop.Sys.TimeSpec timeSpec = default;
+
+            long seconds = time.ToUnixTimeSeconds();
+            long nanoseconds = UnixTimeSecondsToNanoseconds(time, seconds);
+
+            timeSpec.TvSec = seconds;
+            timeSpec.TvNsec = nanoseconds;
+
+            Interop.libc.AttrList attrList = default;
+            attrList.bitmapCount = Interop.libc.AttrList.ATTR_BIT_MAP_COUNT;
+            attrList.commonAttr = Interop.libc.AttrList.ATTR_CMN_CRTIME;
+
+            Interop.Error error =  (Interop.Error)Interop.libc.setattrlist(path, &attrList, &timeSpec, sizeof(Interop.Sys.TimeSpec), default(CULong));
```

Comment:
[setattrlist](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/setattrlist.2.html) can receive FSOPT_NOFOLLOW in the `options` argument:
> If this bit is set, setattrlist() will not follow a sym-link symlink link if it occurs as the last component of path.

without the flag, I think this will set the creation date at the target in case of symlink. Is that consistent with the other dates? I know we have an open issue for the inconsistencies but wanted to mention it for awareness.

Reply:
Yes, it was split according to https://github.com/dotnet/runtime/pull/49555#issuecomment-904413721. Good thing to point.

Reply:
Specific commit to reverse in the other PR is https://github.com/hamarb123/runtime/pull/49555/commits/f84293583bb0c8e1060bdacfb80158a6c64c5fbd (will need modification of multiple things though).